tests/flattenTest.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/**
+ * @covers Dash\flatten
+ */
+class flattenTest extends PHPUnit_Framework_TestCase
+{
+	/**
+	 * @dataProvider cases
+	 */
+	public function test($iterable, $expected)
+	{
+		$this->assertSame($expected, Dash\flatten($iterable));
+	}
+
+	public function cases()
+	{
+		return [
+			'With nulls' => [
+				'iterable' => [null, null, null],
+				'expected' => [null, null, null],
+			],
+			'With empty iterable' => [
+				'iterable' => [],
+				'expected' => [],
+			],
+
+			/*
+				With indexed array
+			 */
+
+			'With an indexed array with only scalar values' => [
+				'iterable' => [
+					6,
+					1,
+					3,
+				],
+				'expected' => [6, 1, 3]
+			],
+			'With an indexed array with a mix of scalar and array values' => [
+				'iterable' => [
+					4,
+					[2, 3],
+					[7,9],
+					8,
+				],
+				'expected' => [4, 2, 3, 7, 9, 8]
+			],
+			'With an indexed array with only array values' => [
+				'iterable' => [
+					[2, 3],
+					[7,9],
+				],
+				'expected' => [2, 3, 7, 9]
+			],
+
+			/*
+				With associative array
+			 */
+
+			'With an associative array with only scalar values' => [
+				'iterable' => [
+					'a' => 6,
+					'b' => 1,
+					'c' => 3,
+				],
+				'expected' => [ 6, 1, 3]
+			],
+			'With an associative array with a mix of scalar and array values' => [
+				'iterable' => [
+					'a' => 4,
+					'b' => [2, 3],
+					'c' => [7,9],
+					'd' => 8,
+				],
+				'expected' => [4, 2, 3, 7, 9, 8]
+			],
+			'With an associative array with only array values' => [
+				'iterable' => [
+					'b' => [2, 3],
+					'c' => [7,9],
+				],
+				'expected' => [2, 3, 7, 9]
+			],
+
+			/*
+				With stdClass
+			 */
+
+			'With an stdClass with only scalar values' => [
+				'iterable' => (object) [
+					'a' => 6,
+					'b' => 1,
+					'c' => 3,
+				],
+				'expected' => [6, 1, 3]
+			],
+			'With an stdClass with a mix of scalar and array values' => [
+				'iterable' => (object) [
+					'a' => 4,
+					'b' => [2, 3],
+					'c' => [7,9],
+					'd' => 8,
+				],
+				'expected' => [4, 2, 3, 7, 9, 8]
+			],
+			'With an stdClass with only array values' => [
+				'iterable' => (object) [
+					'b' => [2, 3],
+					'c' => [7,9],
+				],
+				'expected' => [2, 3, 7, 9]
+			],
+
+			/*
+				With ArrayObject
+			 */
+
+			'With an ArrayObject with only scalar values' => [
+				'iterable' => new ArrayObject([
+					'a' => 6,
+					'b' => 1,
+					'c' => 3,
+				]),
+				'expected' => [6, 1, 3]
+			],
+			'With an ArrayObject with a mix of scalar and array values' => [
+				'iterable' => new ArrayObject([
+					'a' => 4,
+					'b' => [2, 3],
+					'c' => [7,9],
+					'd' => 8,
+				]),
+				'expected' => [4, 2, 3, 7, 9, 8]
+			],
+			'With an ArrayObject with only array values' => [
+				'iterable' => new ArrayObject([
+					'b' => [2, 3],
+					'c' => [7,9],
+				]),
+				'expected' => [2, 3, 7, 9]
+			]
+		];
+	}
```

Comment:
Going forward, don’t feel that you need to exactly copy the same test method structure as before. If it’s better, feel free to use the explicitly-named one-off test methods that you’ve taught us.